crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -157,26 +162,27 @@ impl RuffSettingsIndex {
 
                 // If the directory is excluded from the workspace, skip it.
                 if let Some(file_name) = directory.file_name() {
-                    if let Some((_, settings)) = index
+                    let settings = index
                         .read()
                         .unwrap()
                         .range(..directory.clone())
                         .rfind(|(path, _)| directory.starts_with(path))
-                    {
-                        if match_exclusion(&directory, file_name, &settings.file_resolver.exclude) {
-                            tracing::debug!("Ignored path via `exclude`: {}", directory.display());
-                            return WalkState::Continue;
-                        } else if match_exclusion(
-                            &directory,
-                            file_name,
-                            &settings.file_resolver.extend_exclude,
-                        ) {
-                            tracing::debug!(
-                                "Ignored path via `extend-exclude`: {}",
-                                directory.display()
-                            );
-                            return WalkState::Continue;
-                        }
+                        .map(|(_, settings)| settings.clone())
+                        .unwrap_or_else(|| fallback.clone());
+
+                    if match_exclusion(&directory, file_name, &settings.file_resolver.exclude) {
+                        tracing::debug!("Ignored path via `exclude`: {}", directory.display());
+                        return WalkState::Continue;
```

Comment:
I'm a bit surprised why we return `Continue` but the comment at the top says: *If the directory is excluded from the workspace, skip it*

Reply:
Oh my! Good catch. I think I confused `Skip` with `Continue` in https://github.com/astral-sh/ruff/pull/12299 🤦‍♂️